contest/contest_info.py

```diff
@@ -339,9 +339,6 @@ def user_can_register_contest(user, contest):
     has_ownership = has_contest_ownership(user,contest)
     if has_ownership:
         return False
```

Comment:
By the way, I realize that it's not a good idea to allow public users register contests.
How about add ```if is_public_user(user): return False```